internal/resource/config/token_processor_to_token_generator_mapping_resource.go

```diff
@@ -0,0 +1,334 @@
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type tokenProcessorToTokenGeneratorMappingsResourceModel struct {
+	AttributeSources                 types.List   `tfsdk:"attribute_sources"`
+	AttributeContractFulfillment     types.Map    `tfsdk:"attribute_contract_fulfillment"`
+	IssuanceCriteria                 types.Object `tfsdk:"issuance_criteria"`
+	SourceId                         types.String `tfsdk:"source_id"`
+	TargetId                         types.String `tfsdk:"target_id"`
+	Id                               types.String `tfsdk:"id"`
+	CustomId                         types.String `tfsdk:"custom_id"`
+	DefaultTargetResource            types.String `tfsdk:"default_target_resource"`
+	LicenseConnectionGroupAssignment types.String `tfsdk:"license_connection_group_assignment"`
+}
+
+// GetSchema defines the schema for the resource.
+func (r *tokenProcessorToTokenGeneratorMappingsResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Manages Token Processor To Token Generator Mappings",
+		Attributes: map[string]schema.Attribute{
+			"custom_id": schema.StringAttribute{
+				Description: "The ID of the token processor to token generator mapping. The ID cannot be modified once the instance is created. Note: Ignored when specifying a connection's adapter override.",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"attribute_contract_fulfillment": attributecontractfulfillment.AttributeContractFulfillmentSchema(true),
+			"attribute_sources":              attributesources.AttributeSourcesSchema(),
+			"default_target_resource": schema.StringAttribute{
+				Description: "Default target URL for this Token Processor to Token Generator mapping configuration.",
+				Optional:    true,
+			},
+			"license_connection_group_assignment": schema.StringAttribute{
+				Description: "The license connection group.",
+				Optional:    true,
+			},
+			"target_id": schema.StringAttribute{
+				Description: "The id of the Token Generator.",
+				Required:    true,
+			},
+			"source_id": schema.StringAttribute{
+				Description: "The id of the Token Processor.",
+				Required:    true,
+			},
+			"issuance_criteria": issuancecriteria.IssuanceCriteriaSchema(),
+		},
+	}
+	AddCommonSchema(&schema)
+	resp.Schema = schema
+}
+
+func addOptionalTokenProcessorToTokenGeneratorMappingFields(ctx context.Context, addRequest *client.TokenToTokenMapping, plan tokenProcessorToTokenGeneratorMappingsResourceModel) error {
+	var err error
+	if internaltypes.IsDefined(plan.AttributeSources) {
+		attributeSourcesAttr := plan.AttributeSources.Elements()
+		addRequest.AttributeSources = []client.AttributeSourceAggregation{}
+		for _, source := range attributeSourcesAttr {
```

Comment:
We could move this logic to common, right? Same for the issuance criteria section below?